### DISCUSSION TITLE
Anisotropic Minimum Dissipation buoyancy modification term

### DISCUSSION THREAD

**Discussion by SamDeAbreu** [CliMA/Oceananigans.jl#3843] [Computational science!] [answered]
Hi everyone!

I wanted to ask why the AMD closure doesn't have the "proper" implementation of the buoyancy modification term in the eddy viscosity (subtracting the mean buoyancy field from the buoyancy). I've also seen discussion on here that even if it were correctly implemented, it likely wouldn't increase the closure's accuracy (e.g., https://github.com/CliMA/Oceananigans.jl/issues/1381#issue-811448040)? Is the original implementation (Cb=0) just more accurate in general for stratified flows? I'm relatively new to LES closures, so any information would be greatly appreciated!

**Comment by iuryt** [comment]
I am not familiar with the AMD formulation and sorry if this is not related to what you asked, but afaik the buoyancy field in Oceananigans implementation is already the buoyancy anomaly. Other developers can confirm that.

**Reply by glwagner** [reply]
I think what @SamDeAbreu is referring to is that equation (9) in [Abkar and Moin](https://link.springer.com/article/10.1007/s10546-017-0288-4) involves the potential temperature anomaly. Honestly though I am now a little confused as to whether this term is simply buoyancy. We can't just subtract the horizontally-averaged buoyancy (or any other buoyancy term) and get the same dynamics.

**Reply by SamDeAbreu** [reply]
Looking at Abkar and Moin (2017), the theta minus horizontally averaged theta is simply their buoyancy term in their equations of motion. Doesn't that mean then that the `g*beta*(theta - <theta>)` term in the eddy viscosity is just buoyancy, implying no need to subtract a horizontally component of the buoyancy to begin with?

**Reply by glwagner** [reply]
Hmm yeah might be. We can't just subtract a horizontally-averaged term from the buoyancy; this would have to be accounted for elsewhere...

Come to think of it, you may be able to easily test the effect of removing the horizontal average using `BackgroundField`.

**Comment by glwagner** [comment, marked as answer]
Let me start with a little background. AMD is "just" a model for turbulence --- despite the pretensions of its "derivation" --- and as such should be regarded like any other hypothesis that needs to be verified. As far as I know, there is not extensive evidence that this "buoyancy modification" to the SGS viscosity is actually important. But if you have more information please provide it.

In very limited testing, it doesn't seem to make much of a difference. I would encourage you to do some testing yourself as well (if possible, because this may require modifying the source code). You can try setting up problems with some mean buoyancy gradient and manually subtracting it from the buoyancy estimated inside AMD for some limited number of tests. If you find a difference (and really, if you find that using proper buoyancy makes your simulations more accurate), then you have some cause to move forward...

It makes sense to me that this term has only a minor effect, because like many SGS models we find that _in practice_ we can't just use AMD at any resolution: we need a high enough resolution so that motions down to the Ozmidov scale are resolved. Thus motions at the grid scale --- which dominate the gradients that enter into the SGS model --- are nearly isotropic and not strongly affected by buoyancy.

From a more practical point of view, it is not that easy to design an approach that can consistently subtract a large-scale field from the buoyancy for all situations. The easiest case is horizontally-homogeneous LES. But LES can be run in any configuration at all. It's always best to follow the prescription "do no harm". In other words it's better to make no approximations then to implement wrong approximations based on assumptions about what users are trying to do.

That said, it is possible in principle to design an interface whereby users might manually specify the kind of buoyancy they would like to include in AMD.

For my own LES (which is limited, to be fair), I have started using purely implicit diffusion with `WENO(order=9)` and `closure=nothing`. We find this scheme has a higher effective resolution than trying to put in an explicit closure --- it can yield similar results with half as many grid points in each direction (8x smaller), and in extreme cases 1/4 the grid points in each direction (64x smaller simulation!) So it makes simulations cheaper (sometimes way cheaper), and can get you to science results faster depending on the kind of science you are trying to do. 

[Pressel et al 2017](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1002/2016MS000778) presents some nice arguments for why implicit dissipation is a better choice, at least in terms of effective resolution.

If you're interested in learning more about closures for LES I suggest you get in touch on #3638 or #3642, helping out with testing some of those new closures could teach you a bit about the closure problem.

@johnryantaylor or @jagoosw or @tomchor may have more to add.

**Reply by SamDeAbreu** [reply]
Thank you very much for the detailed response! It didn't occur to me that if one resolves the Ozmidov scale then the motion is essentially isotropic at SGS, which renders the term not important, as you note. The fact that I'm seeing a notable difference between Cb=1 and Cb=0 is probably implying I don't have a sufficient resolution for Cb=0 to begin with. 

No closure with WENO advection is interesting. One of my goals is to compute a TKE budget. Would you happen to know how one could compute the dissipation WENO advection is implicitly inducing, or is talking about energetics with no closure and a dissipative advection scheme a bit of a gray area?

**Reply by glwagner** [reply]
There is no grey area regarding energetics. However, we don't have a formula to compute dissipation (though it might in fact be possible in principle to derive one). When we use dissipative advection we actually have to compute dissipation _using_ the residual of the kinetic energy budget.

@simone-silvestri has been working on this problem. I think he has developed a prototype diagnostic for dissipation but it is not yet part of Oceananigans.
